src/lstm_layer.cpp

```diff
@@ -1197,6 +1274,23 @@ void LSTM::forward(BaseHiddenStates &input_states,
     output_states.block_size = batch_size;
     output_states.actual_size = this->output_size * this->seq_len;
 
+    // TODO: This is not efficient for memory and performance. Update the
```

Comment:
Can you delete from line 1277 to line 1292? 
With these lines of code, you are propagating the priors (hidden and cell states).
Actually we want to propagate the posteriors. We can achieve that by deleting these lines, because the posteriors are already saved in lstm_states.mu_h_prev, ...  in the backward pass.

Reply:
@van-dai-vuong 
- At first, I think we can remove it but then I realized that if we remove it, it wont work for the test set where we do not perform the backward pass. 
- I have to move the update step toward the end of the backward function because parameter update requires `mu_c_prev` at the current step meaning before updating `mu_c_prev` (see `.bacward(...)`)
- Increasing number of epochs works better
Let me know what do you think

Reply:
Hi @lhnguyen102, thank you! I think it's good now.

Reply:
One other thing is that you need to create a separate issue to clear out all the prior and previous cell states to zero at the end of each epoch during the training